### PR TITLE
Lazily register resources without reading the file system

### PR DESCRIPTION
**Pull request by joelvh**
By removing the need to read from the file system, all dependencies can be packaged into one file with things like Webpack.

### CODE REVIEW

**Comment by lpinca** [comment]
See https://github.com/MONEI/Shopify-api-node/issues/88.

**Comment by joelvh** [comment]
I’ve worked around it until now, this update just removes the file system dependency. Acceptable?

**Comment by lpinca** [comment]
I think it complicates things as we have to update the map every time we add a new resource.

**Comment by joelvh** [comment]
I thought about that, but how often is that? You have to add the resource file at that point, as well as  tests. So it would be natural to add a line to the resources as a 3rd place to be updated.

**Comment by lpinca** [comment]
Counter point, how often is this module bundled and why would you do it?

**Comment by joelvh** [comment]
Serverless. All our integrations are going the serverless route, in this case using Lambda functions. There’s a 50mb file size limit. But regardless, the way to package and upload is to use Webpack. We have many services that leverage this package.

I had thought about creating a way to eagerly load each resource as well, so we can package only what we need. That would be a second phase, assuming this is going to be accepted. (Basically, adding the ability to import each resource I want and “register” individually.)

**Comment by lpinca** [comment]
Let me think about it (I'm not a fan of the resources map, I would rather remove the lazy load functionality).

**Comment by joelvh** [comment]
Agreed. Without lazy loading, we could have a way to require resources individually, but would still need to add new require statements for new resources.

**Comment by lpinca** [comment]
Thank you.

**Comment by joelvh** [comment]
Thanks @lpinca ! Would you accept a pull request with a refactoring to make lazy loading optional?

**Comment by lpinca** [comment]
If it turns out to be small and clean I'm fine with it. Let's also keep open the option of completely removing the lazy loading feature (it would probably be a breaking change).

File: resources/index.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  Object.keys(RESOURCES).forEach(prop => {
+    Object.defineProperty(Shopify.prototype, prop, {
+      get: function get() {
+        const resource = require(`./${RESOURCES[prop]}`);
+
+        return Object.defineProperty(this, prop, {
+          value: new resource(this)
+        })[prop];
+      },
+      set: function set(value) {
```
Comment (joelvh) [inline resources/index.js]: @lpinca reviewing this, does the `value` ever get used after being set since the getter always instantiates a value?

Comment (lpinca) [reply resources/index.js]: Yes, see https://github.com/MONEI/Shopify-api-node/blob/master/test/shopify.test.js#L89-L99.

Comment (lpinca) [reply resources/index.js]: After being set is will be on the object, not the prototype so the get is no longer used.

Comment (joelvh) [reply resources/index.js]: Ok makes sense. Thanks.

**Pull request merged**